ocaml/xapi/xapi_pif.ml

```diff
@@ -813,8 +813,23 @@ let rec unplug ~__context ~self =
     debug "PIF is network SRIOV physical PIF, also bringing down SRIOV logical PIF";
     let pif = Db.Network_sriov.get_logical_PIF ~__context ~self:(List.hd sriov)in
     unplug ~__context ~self:pif
+    (* If this PIF is also a bond slave, it will be checked later to make sure that 
+         * this bond slave will not be brought down here *)
   end;
-  Nm.bring_pif_down ~__context self
+  let bonds = Db.PIF.get_bond_master_of ~__context ~self in
```

Comment:
Can you use the pif topology to get the bond and check if this is a SRIOV bond slave?

Reply:
@minglumlu Agree with Marcello, and you can refine the code of ` if Db.PIF.get_sriov_logical_PIF_of ~__context ~self <> [] then begin` above.

Reply:
Hi @mseri 
Do you expect a new return of `get_pif_topo` for something like `bond :: Network_sriov_physical :: _`?

And `get_pif_topo` will call `get_pif_type` multiple times, in one of which checking on `tunnel`, `vlan`, and `sriov` will be done also. So I think it might be more directly to check the two fields here.

Reply:
Sure.